Solutions/Azure Active Directory/Analytic Rules/BruteForceCloudPC.yaml

```diff
@@ -27,21 +27,20 @@ query: |
   | where AppDisplayName =~ "Windows Sign In"
   // Split out failure versus non-failure types
   | extend FailureOrSuccess = iff(ResultType in ("0", "50125", "50140", "70043", "70044"), "Success", "Failure")
-  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), IPAddress = makeset(IPAddress), makeset(OS), makeset(Browser), makeset(City), 
-  makeset(ResultType), FailureCount = countif(FailureOrSuccess=="Failure"), SuccessCount = countif(FailureOrSuccess=="Success") 
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), IPAddress = make_set(IPAddress,100), makeset(OS,20), make_set(Browser,20), make_set(City,100), make_set(ResultType,100), FailureCount = countif(FailureOrSuccess=="Failure"), SuccessCount = countif(FailureOrSuccess=="Success") 
   by bin(TimeGenerated, authenticationWindow), UserDisplayName, UserPrincipalName, AppDisplayName
   | where FailureCount >= failureCountThreshold and SuccessCount >= successCountThreshold
-  | mvexpand IPAddress
+  | mv-expand IPAddress
   | extend IPAddress = tostring(IPAddress)
-  | extend timestamp = StartTime, AccountCustomEntity = UserPrincipalName, IPCustomEntity = IPAddress
+  | extend timestamp = StartTime
 entityMappings:
   - entityType: Account
     fieldMappings:
       - identifier: FullName
-        columnName: AccountCustomEntity
+        columnName: UserPrincipalName
```

Comment:
Fullname is not a strong entity identifier for Account (its actually a legacy mapping) where we have a UPN we should split in into Name and UPN suffix and have both as entities - https://learn.microsoft.com/en-us/azure/sentinel/entities-reference